### PR TITLE
[Product Refactor] Remove multiple taxons

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Related to #9069 

Removes a bunch of unused code and functionality around Taxons. This is a hangover from Spree, where theoretically a product could have multiple taxons (product categories). We only have **one** taxon per product, there's no way to actually set more than one.

This is a bit of prep for migrating taxons from products to variants (we're keeping only the `primary_taxon_id` and removing this weird mess around multiple taxons per product).

Review note: this unused "multiple taxons" functionality is handled via a class called `Spree::Classification` which represents a join table called `spree_products_taxons`.

#### What should we test?

Not sure what can be manually tested, we're just removing some unused functionality here.

DC: Still, it seems important enough to manually verify the existing behaviour. Just a basic check for a single product should be fine:

1. Add/update category on a product
2. Try filtering products by category as a customer
3. Check that it appears in a report

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/components/product_component_spec.rb [hunk 1]
```diff
@@ -35,12 +35,6 @@
         )
       )
     end
-
-    it "joins the categories' name" do
-      expect(page.find('.category')).to have_content(
-        /random 1, random 2/, exact: true, normalize_ws: true
-      )
-    end
```
Comment (dacook) [inline spec/components/product_component_spec.rb:43]: Actually this is the only example in the block, so the whole block can be deleted 🧹

File: spec/serializers/api/enterprise_shopfront_serializer_spec.rb [hunk 2]
```diff
@@ -67,6 +67,5 @@
   it "serializes taxons" do
     expect(serializer.serializable_hash[:taxons]).to be_a ActiveModel::ArraySerializer
     expect(serializer.serializable_hash[:taxons].to_json).to match 'Meat'
-    expect(serializer.serializable_hash[:taxons].to_json).to match 'Veg'
```
Comment (dacook) [inline spec/serializers/api/enterprise_shopfront_serializer_spec.rb:70]: At first I was thinking "hey, there shouldn't be multiple taxons anymore."
But a shopfront has multiple products, so of course continues to have multiple taxons 👍

File: spec/system/admin/reports_spec.rb [hunk 3]
```diff
@@ -402,8 +402,6 @@
     before do
       product1.set_property 'Organic', 'NASAA 12345'
       product2.set_property 'Organic', 'NASAA 12345'
-      product1.taxons = [taxon]
-      product2.taxons = [taxon]
```
Comment (dacook) [inline spec/system/admin/reports_spec.rb:406]: Hum, this spec works on the factory data for primary taxon. 
Shrug, not worth  updating.

**Review by dacook — CHANGES_REQUESTED** [review]
Great job, a nice tidy-up.

I see even more more code that can be deleted, so will go ahead.

**Review by dacook — APPROVED** [review]

**Review by dacook — CHANGES_REQUESTED** [review]
I got trigger-happy and missed the linter error:

```
app/models/spree/taxon.rb:8:5: C: Rails/HasManyOrHasOneDependent: Specify a `:dependent` option.
```

The Products class says it requires a primary_taxon to be present, but the Taxon class doesn't specify what to do when it is deleted.
The options are here:
https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many-label-Options

We could explicitly specify the current behaviour:
`dependent: nil`

But I think it makes more sense to `:restrict_with_error` (causes an error to be added to the owner if there are any associated objects). What do you think?

**Comment by dacook** [comment]
PS I found that I wasn't able to revert my dev environment so updated the migration to make it easy.

**Review by rioug — APPROVED** [review]
I am always happy to see less code :1st_place_medal:

**Comment by Matt-Yorkley** [comment]
> I think it makes more sense to :restrict_with_error

Yeah nice catch, I think similar behaviour was provided before through the `Classification` class. I'll add it now :+1:

**Review by dacook — APPROVED** [review]
Cool 👍

**Comment by dacook** [comment]
Regarding testing: I think it would be worth some manual testing around categories. There _shouldn't_ be any visible changes, but I'm not confident enough to be able to guarantee it. I think it's important enough to manually verify.
Not being very familiar with categories, I can only suggest the basics, ie:

1. Add/update category on a product
2. Try filtering products by category as a customer
3. Check that it appears in a report

I'll add the above to the description, but @Matt-Yorkley feel free to chime in if this is really unnecessary.

**Comment by drummer83** [comment]
Hi @Matt-Yorkley,
Thanks for working on this! I have tested on staging AU.

I can confirm the following:
- Changing taxons of a product as enterprise user works as before. :heavy_check_mark: 
- Creating, editing and deleting taxons as super admin user works as before. :heavy_check_mark: 
image
- Filtering in the shopfront and checking out works as before. :heavy_check_mark: 
image
- Taxons are showing up in the reports as before. :heavy_check_mark: 
image

All seems to be good. :partying_face: 
Merging! :rocket: 
Thanks!

**Pull request merged**